### ISSUE TITLE
kt_android_library plugin not working

### ISSUE THREAD

**Issue by act262** [bazelbuild/rules_kotlin#180]
I use `kt_android_library` with plugins to process Annotation processor, but it  not working.

But while using `android_library` or `android_binary` it working well.

what's wrong here ?

```
kt_android_library(
    name = 'wrapper',

    # package name
    custom_package ='io.mm',

    # AndroidManifest
    manifest = "src/main/AndroidManifest.xml",

    # java + kotlin sources
    srcs = glob(['src/main/java/**',]),

    # res
    resource_files = glob(['src/main/res/**',]),

    # apt
    plugins = ['aptPlugin'],
    .......
)

# Annotation processor
java_import(
    name = 'aptDeps',
    jars = glob(['build/bazel/apt/**/*.jar'])
)

java_plugin(
    name = 'aptPlugin',
    generates_api = 1,
    deps = ['aptDeps'],

    processor_class = 'com.alibaba.android.arouter.compiler.processor.RouteProcessor',
)

```

**Comment by cgruber** [comment]
What's the error?  It's possible that you're having an issue with rules_kotlin bug which doesn't properly forward the compile-time deps.  But I'd need to see the error to know.

**Comment by act262** [comment]
There is no error , just annotationProcessor didn't execute normally. 
I think `kt_android_library` maybe not full delegate it attrs

**Comment by nkoroste** [comment]
Got same issue!
The jar is basically is not on the host path. When it works properly there is a file `bazel-out/host/bin/platform/config/processor/liblib.jar` however, when `kt_android_library` is used this jar is listed in the params file but the jar is not there.

**Comment by cgruber** [comment]
What version? We're using dagger via the plugin system in kt_android_library calls inside square, using the recent legacy-1.3.0 prerelease, and have been for a long time.

**Comment by nkoroste** [comment]
rules_kotlin_version = "legacy-1.3.0-rc1"
and Dagger v2.24

**Comment by nkoroste** [comment]
FYI: This is still an issue in **rules_kotlin_version = "legacy-1.3.0-rc3"**

Some more info, in my case I have the following 2 plugins and 2 targets that export those plugins:
```
java_plugin(
    name = "dagger_with_compiler_plugin",
    generates_api = True,
    processor_class = "dagger.internal.codegen.ComponentProcessor",
    deps = ["@host_mvn//:com_google_dagger_dagger_compiler"],
)

java_library(
    name = "dagger_with_compiler",
    exported_plugins = [":dagger_with_compiler_plugin"],
    visibility = ["//visibility:public"],
    exports = [
        "@android_mvn//:com_google_dagger_dagger",
        # HACK
        #        "@host_mvn//:com_google_dagger_dagger_compiler",
    ],
)
```
```
java_plugin(
    name = "dagger_android_plugin",
    generates_api = True,
    processor_class = "dagger.android.processor.AndroidProcessor",
    deps = [
        "@host_mvn//:com_google_dagger_dagger_android_processor",
        "@host_mvn//:com_google_dagger_dagger_compiler",
    ],
)

java_library(
    name = "dagger_android",
    exported_plugins = [":dagger_android_plugin"],
    visibility = ["//visibility:public"],
    exports = [
        "@android_mvn//:com_google_dagger_dagger_android",
        "@android_mvn//:com_google_dagger_dagger_android_support",
        # HACK
        #        "@host_mvn//:com_google_dagger_dagger_android_processor",
        #        "@host_mvn//:com_google_dagger_dagger_compiler",
    ],
)
```

When I try to consume the above libs that export the plugins in a `kt_android_library` it fails because it cannot find:
```
warning: [kapt] Can't find annotation processor class dagger.android.processor.AndroidProcessor: dagger.android.processor.AndroidProcessor
warning: [kapt] Can't find annotation processor class dagger.internal.codegen.ComponentProcessor: dagger.internal.codegen.ComponentProcessor
```
(side note: Why is this a "**warning**"? it should be an **Error**)
And therefore, fails to generate my dagger files `DaggerApplicationComponent`.

**Further debugging:**
Inspecting the generated `targetName_kt.jar-0.params` I can see that the above processors are in fact listed defined correctly and their jars are listed under `--processorpath`
```
--processors
dagger.android.processor.AndroidProcessor
dagger.internal.codegen.ComponentProcessor
--processorpath
bazel-out/host/bin/external/host_mvn/v1/gcs/snapengine-maven-publish/releases/com/google/dagger/dagger-android-processor/2.24-jetified-834b21b.206/stamped_dagger-android-processor-2.24-jetified-834b21b.206.jar
bazel-out/host/bin/external/host_mvn/v1/gcs/snapengine-maven-publish/releases/com/google/dagger/dagger-compiler/2.24-jetified-834b21b.206/stamped_dagger-compiler-2.24-jetified-834b21b.206.jar
```
However, if I try to `ls` those jar files I cannot locate them on my machine under that path. Not sure if this is an issue? just an observation at this point. 

I can work around the above issue if I un-comment the deps under "HACK` in the above plugin targets. In this case it will put the processor jars under the class path in the `targetName_kt.jar-0.params` file and the targets will compile successfully.
```
--classpath
bazel-out/k8-fastbuild/bin/external/host_mvn/v1/gcs/snapengine-maven-publish/releases/com/google/dagger/dagger-android-processor/2.24-jetified-834b21b.206/stamped_dagger-android-processor-2.24-jetified-834b21b.206.jar
bazel-out/k8-fastbuild/bin/external/host_mvn/v1/gcs/snapengine-maven-publish/releases/com/google/dagger/dagger-compiler/2.24-jetified-834b21b.206/stamped_dagger-compiler-2.24-jetified-834b21b.206.jar
--processors
dagger.android.processor.AndroidProcessor
dagger.internal.codegen.ComponentProcessor
--processorpath
bazel-out/host/bin/external/host_mvn/v1/gcs/snapengine-maven-publish/releases/com/google/dagger/dagger-android-processor/2.24-jetified-834b21b.206/stamped_dagger-android-processor-2.24-jetified-834b21b.206.jar
bazel-out/host/bin/external/host_mvn/v1/gcs/snapengine-maven-publish/releases/com/google/dagger/dagger-compiler/2.24-jetified-834b21b.206/stamped_dagger-compiler-2.24-jetified-834b21b.206.jar
```
in this case I can actually successfully `ls` the jars listed under `--classpath`. Notice that they're slightly different than the ones under `--processorpath`

That said, I don't think this is the right thing to do putting the dagger processors/compilers on the class path which will basically mean that those jars and their deps will be present at runtime?

**Comment by nkoroste** [comment]
@cgruber any updates on this? 
Seems like `processorpath` is broken.I found another issue that is related to this were guava-jre and guava-android are both present `processorpath` but only the android one is in the `classpath` and thus I get this error:

```
ERROR: /home/nk/Snapchat/Dev/android/snapchat/components/shake2report/core/BUILD.bazel:56:1: Compiling Kotlin to JVM //components/shake2report/core:lib_kt { kt: 99, java: 26, srcjars: 0 } failed (Exit 2)
warning: some JAR files in the classpath have the Kotlin Runtime library bundled into them. This may cause difficult to debug problems if there's a different version of the Kotlin Runtime library in the classpath. Consider removing these libraries from the classpath
bazel-out/k8-fastbuild/bin/external/android_mvn/v1/https/jcenter.bintray.com/org/jetbrains/kotlin/kotlin-stdlib/1.3.60/stamped_kotlin-stdlib-1.3.60.jar: warning: library has Kotlin runtime bundled into it
error: components/shake2report/core/src/main/java/com/snap/shake2report/config/Shake2ReportConfigModule.java:8: error: cannot find symbol
        Shake2ReportConfigModule_ConfigModule.class,
        ^
  symbol: class Shake2ReportConfigModule_ConfigModule
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
exception: java.lang.IllegalStateException: failed to analyze: java.lang.reflect.InvocationTargetException
	at org.jetbrains.kotlin.analyzer.AnalysisResult.throwIfError(AnalysisResult.kt:56)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:182)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:165)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:55)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:84)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:42)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:104)
	at io.bazel.kotlin.compiler.BazelK2JVMCompiler.exec(BazelK2JVMCompiler.kt:29)
	at jdk.internal.reflect.GeneratedMethodAccessor85.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.bazel.kotlin.builder.toolchain.KotlinToolchain$KotlinCliToolInvoker.compile(KotlinToolchain.kt:124)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.runAnnotationProcessor(KotlinJvmTaskExecutor.kt:96)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.runAnnotationProcessors(KotlinJvmTaskExecutor.kt:131)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.access$runAnnotationProcessors(KotlinJvmTaskExecutor.kt:38)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor$preProcessingSteps$2.invoke(KotlinJvmTaskExecutor.kt:61)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor$preProcessingSteps$2.invoke(KotlinJvmTaskExecutor.kt:38)
	at io.bazel.kotlin.builder.utils.CompilationTaskContext.execute(CompilationTaskContext.kt:140)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.preProcessingSteps(KotlinJvmTaskExecutor.kt:59)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.execute(KotlinJvmTaskExecutor.kt:45)
	at io.bazel.kotlin.builder.tasks.KotlinBuilder.executeJvmTask(KotlinBuilder.kt:186)
	at io.bazel.kotlin.builder.tasks.KotlinBuilder.apply(KotlinBuilder.kt:51)
	at io.bazel.kotlin.builder.tasks.BazelWorker.runAsPersistentWorker(BazelWorker.kt:83)
	at io.bazel.kotlin.builder.tasks.BazelWorker.apply(BazelWorker.kt:64)
	at io.bazel.kotlin.builder.KotlinBuilderMain.main(KotlinBuilderMain.java:30)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jetbrains.kotlin.kapt3.base.AnnotationProcessingKt.doAnnotationProcessing(annotationProcessing.kt:76)
	at org.jetbrains.kotlin.kapt3.base.AnnotationProcessingKt.doAnnotationProcessing$default(annotationProcessing.kt:35)
	at org.jetbrains.kotlin.kapt3.AbstractKapt3Extension.runAnnotationProcessing(Kapt3Extension.kt:224)
	at org.jetbrains.kotlin.kapt3.AbstractKapt3Extension.analysisCompleted(Kapt3Extension.kt:187)
	at org.jetbrains.kotlin.kapt3.ClasspathBasedKapt3Extension.analysisCompleted(Kapt3Extension.kt:98)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM$analyzeFilesWithJavaIntegration$2.invoke(TopDownAnalyzerFacadeForJVM.kt:97)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:107)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:82)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:557)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:82)
	at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:107)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:548)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:177)
	... 23 more
Caused by: com.sun.tools.javac.processing.AnnotationProcessingError: java.lang.NoSuchMethodError: com.google.common.collect.ImmutableList.toImmutableList()Ljava/util/stream/Collector;
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:984)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:888)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1214)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1326)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1258)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1157)
	... 40 more
Caused by: java.lang.NoSuchMethodError: com.google.common.collect.ImmutableList.toImmutableList()Ljava/util/stream/Collector;
	at dagger.internal.codegen.binding.InjectionAnnotations.getQualifiers(InjectionAnnotations.java:95)
	at dagger.internal.codegen.validation.DependencyRequestValidator.checkQualifiers(DependencyRequestValidator.java:89)
	at dagger.internal.codegen.validation.DependencyRequestValidator.validateDependencyRequest(DependencyRequestValidator.java:67)
	at dagger.internal.codegen.validation.InjectValidator.validateDependencyRequest(InjectValidator.java:244)
	at dagger.internal.codegen.validation.InjectValidator.validateField(InjectValidator.java:205)
	at dagger.internal.codegen.validation.InjectValidator.validateMembersInjectionType(InjectValidator.java:256)
	at dagger.internal.codegen.validation.InjectBindingRegistryImpl.tryRegisterMembersInjectedType(InjectBindingRegistryImpl.java:280)
	at dagger.internal.codegen.validation.InjectBindingRegistryImpl.tryRegisterMembersInjectedType(InjectBindingRegistryImpl.java:264)
	at dagger.internal.codegen.InjectProcessingStep$1.visitVariableAsField(InjectProcessingStep.java:54)
	at dagger.internal.codegen.InjectProcessingStep$1.visitVariableAsField(InjectProcessingStep.java:44)
	at java.compiler@11.0.1/javax.lang.model.util.ElementKindVisitor6.visitVariable(ElementKindVisitor6.java:237)
	at jdk.compiler/com.sun.tools.javac.code.Symbol$VarSymbol.accept(Symbol.java:1576)
	at dagger.internal.codegen.InjectProcessingStep.process(InjectProcessingStep.java:76)
	at dagger.internal.codegen.validation.TypeCheckingProcessingStep.lambda$process$0(TypeCheckingProcessingStep.java:51)
	at java.base/java.util.Map.forEach(Map.java:661)
	at dagger.internal.codegen.validation.TypeCheckingProcessingStep.process(TypeCheckingProcessingStep.java:48)
	at dagger.internal.codegen.validation.TypeCheckingProcessingStep.process(TypeCheckingProcessingStep.java:34)
	at dagger.internal.codegen.statistics.DaggerStatisticsCollectingProcessingStep.process(DaggerStatisticsCollectingProcessingStep.java:52)
	at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:330)
	at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:181)
	at org.jetbrains.kotlin.kapt3.base.incremental.IncrementalProcessor.process(incrementalProcessors.kt)
	at org.jetbrains.kotlin.kapt3.base.ProcessorWrapper.process(annotationProcessing.kt:147)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:972)
	... 45 more

Compilation failure: compile phase failed:
warning: some JAR files in the classpath have the Kotlin Runtime library bundled into them. This may cause difficult to debug problems if there's a different version of the Kotlin Runtime library in the classpath. Consider removing these libraries from the classpath
bazel-out/k8-fastbuild/bin/external/android_mvn/v1/https/jcenter.bintray.com/org/jetbrains/kotlin/kotlin-stdlib/1.3.60/stamped_kotlin-stdlib-1.3.60.jar: warning: library has Kotlin runtime bundled into it
error: components/shake2report/core/src/main/java/com/snap/shake2report/config/Shake2ReportConfigModule.java:8: error: cannot find symbol
        Shake2ReportConfigModule_ConfigModule.class,
        ^
  symbol: class Shake2ReportConfigModule_ConfigModule
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
warning: warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX
exception: java.lang.IllegalStateException: failed to analyze: java.lang.reflect.InvocationTargetException
	at org.jetbrains.kotlin.analyzer.AnalysisResult.throwIfError(AnalysisResult.kt:56)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:182)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:165)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:55)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:84)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:42)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:104)
	at io.bazel.kotlin.compiler.BazelK2JVMCompiler.exec(BazelK2JVMCompiler.kt:29)
	at jdk.internal.reflect.GeneratedMethodAccessor85.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.bazel.kotlin.builder.toolchain.KotlinToolchain$KotlinCliToolInvoker.compile(KotlinToolchain.kt:124)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.runAnnotationProcessor(KotlinJvmTaskExecutor.kt:96)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.runAnnotationProcessors(KotlinJvmTaskExecutor.kt:131)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.access$runAnnotationProcessors(KotlinJvmTaskExecutor.kt:38)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor$preProcessingSteps$2.invoke(KotlinJvmTaskExecutor.kt:61)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor$preProcessingSteps$2.invoke(KotlinJvmTaskExecutor.kt:38)
	at io.bazel.kotlin.builder.utils.CompilationTaskContext.execute(CompilationTaskContext.kt:140)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.preProcessingSteps(KotlinJvmTaskExecutor.kt:59)
	at io.bazel.kotlin.builder.tasks.jvm.KotlinJvmTaskExecutor.execute(KotlinJvmTaskExecutor.kt:45)
	at io.bazel.kotlin.builder.tasks.KotlinBuilder.executeJvmTask(KotlinBuilder.kt:186)
	at io.bazel.kotlin.builder.tasks.KotlinBuilder.apply(KotlinBuilder.kt:51)
	at io.bazel.kotlin.builder.tasks.BazelWorker.runAsPersistentWorker(BazelWorker.kt:83)
	at io.bazel.kotlin.builder.tasks.BazelWorker.apply(BazelWorker.kt:64)
	at io.bazel.kotlin.builder.KotlinBuilderMain.main(KotlinBuilderMain.java:30)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jetbrains.kotlin.kapt3.base.AnnotationProcessingKt.doAnnotationProcessing(annotationProcessing.kt:76)
	at org.jetbrains.kotlin.kapt3.base.AnnotationProcessingKt.doAnnotationProcessing$default(annotationProcessing.kt:35)
	at org.jetbrains.kotlin.kapt3.AbstractKapt3Extension.runAnnotationProcessing(Kapt3Extension.kt:224)
	at org.jetbrains.kotlin.kapt3.AbstractKapt3Extension.analysisCompleted(Kapt3Extension.kt:187)
	at org.jetbrains.kotlin.kapt3.ClasspathBasedKapt3Extension.analysisCompleted(Kapt3Extension.kt:98)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM$analyzeFilesWithJavaIntegration$2.invoke(TopDownAnalyzerFacadeForJVM.kt:97)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:107)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:82)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:557)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:82)
	at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:107)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:548)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:177)
	... 23 more
Caused by: com.sun.tools.javac.processing.AnnotationProcessingError: java.lang.NoSuchMethodError: com.google.common.collect.ImmutableList.toImmutableList()Ljava/util/stream/Collector;
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:984)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:888)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1214)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1326)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1258)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1157)
	... 40 more
Caused by: java.lang.NoSuchMethodError: com.google.common.collect.ImmutableList.toImmutableList()Ljava/util/stream/Collector;
	at dagger.internal.codegen.binding.InjectionAnnotations.getQualifiers(InjectionAnnotations.java:95)
	at dagger.internal.codegen.validation.DependencyRequestValidator.checkQualifiers(DependencyRequestValidator.java:89)
	at dagger.internal.codegen.validation.DependencyRequestValidator.validateDependencyRequest(DependencyRequestValidator.java:67)
	at dagger.internal.codegen.validation.InjectValidator.validateDependencyRequest(InjectValidator.java:244)
	at dagger.internal.codegen.validation.InjectValidator.validateField(InjectValidator.java:205)
	at dagger.internal.codegen.validation.InjectValidator.validateMembersInjectionType(InjectValidator.java:256)
	at dagger.internal.codegen.validation.InjectBindingRegistryImpl.tryRegisterMembersInjectedType(InjectBindingRegistryImpl.java:280)
	at dagger.internal.codegen.validation.InjectBindingRegistryImpl.tryRegisterMembersInjectedType(InjectBindingRegistryImpl.java:264)
	at dagger.internal.codegen.InjectProcessingStep$1.visitVariableAsField(InjectProcessingStep.java:54)
	at dagger.internal.codegen.InjectProcessingStep$1.visitVariableAsField(InjectProcessingStep.java:44)
	at java.compiler@11.0.1/javax.lang.model.util.ElementKindVisitor6.visitVariable(ElementKindVisitor6.java:237)
	at jdk.compiler/com.sun.tools.javac.code.Symbol$VarSymbol.accept(Symbol.java:1576)
	at dagger.internal.codegen.InjectProcessingStep.process(InjectProcessingStep.java:76)
	at dagger.internal.codegen.validation.TypeCheckingProcessingStep.lambda$process$0(TypeCheckingProcessingStep.java:51)
	at java.base/java.util.Map.forEach(Map.java:661)
	at dagger.internal.codegen.validation.TypeCheckingProcessingStep.process(TypeCheckingProcessingStep.java:48)
	at dagger.internal.codegen.validation.TypeCheckingProcessingStep.process(TypeCheckingProcessingStep.java:34)
	at dagger.internal.codegen.statistics.DaggerStatisticsCollectingProcessingStep.process(DaggerStatisticsCollectingProcessingStep.java:52)
	at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:330)
	at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:181)
	at org.jetbrains.kotlin.kapt3.base.incremental.IncrementalProcessor.process(incrementalProcessors.kt)
	at org.jetbrains.kotlin.kapt3.base.ProcessorWrapper.process(annotationProcessing.kt:147)
	at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:972)
	... 45 more
```

Related: https://github.com/google/dagger/issues/1700

**Comment by kitterion** [comment]
I had a similar issue while trying to set up glide to work, so I did some digging.
The plugin aspect propagates paths to plugins as strings
https://github.com/bazelbuild/rules_kotlin/blob/b40d920c5a5e044c541513f0d5e9260d0a4579c0/kotlin/internal/jvm/plugins.bzl#L52
By changing it to propagate File objects instead, letting bazel deal with extracting paths to them here (thankfully, `args.add_all` happily accepts File objects)
https://github.com/bazelbuild/rules_kotlin/blob/b40d920c5a5e044c541513f0d5e9260d0a4579c0/kotlin/internal/jvm/compile.bzl#L206
and then adding them to the list of inputs here
https://github.com/bazelbuild/rules_kotlin/blob/b40d920c5a5e044c541513f0d5e9260d0a4579c0/kotlin/internal/jvm/compile.bzl#L218
I was able to locally fix the issues I had with the glide annotation processor.
I don't know why the plugins don't end up in `compile_jars`, but apparently they don't, because looking at `aquery` output, the plugin jar was not on the list of inputs to the compile action.
Hopefully, it's enough to fix this bug as well.
